### PR TITLE
[wasm] Update xnnpack

### PR DESCRIPTION
**Pull request by chunnienc**
This PR updates xnnpack and its dependencies' versions.

Upgrade: emsdk is upgraded from 3.1.28 to 3.1.35 to align with the version in google3. Otherwise xnnpack may not compile.

When compiling TFJS wasm kernels with xnnpack with `-c opt`, it throws an unused function error in xnnpack's code base. Therefore this PR introduces a new bazel config `--config=xnnpack-opt` to add flags to disable xnnpack logging feature and further rule out compiling logging codes.

Breaking changes need to be dealt with are reported here #7530 and will be addressed in a separate PR.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7507)

### CODE REVIEW

**Review by pyu10055 — APPROVED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7507)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @mattsoulanille)

**Review by mattsoulanille — APPROVED** [review]
LGTM with a nit

File: WORKSPACE [hunk 1]
```diff
@@ -102,9 +102,8 @@ http_archive(
     name = "emsdk",
     # TODO: Remove repo_mapping when emsdk updates to rules_nodejs 5
     repo_mapping = {"@nodejs": "@nodejs_host"},
-    sha256 = "b8270749b99d8d14922d1831b93781a5560fba6f7bce65cd477fc1b6aa262535",
-    strip_prefix = "emsdk-3.1.28/bazel",
-    urls = ["https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.28.tar.gz"],
+    strip_prefix = "emsdk-3.1.26/bazel",
+    urls = ["https://github.com/emscripten-core/emsdk/archive/refs/tags/3.1.26.tar.gz"],
```
[the remark this answers is not in the document]
Comment (chunnienc) [reply WORKSPACE:107]: My bad. I was switching between versions and I forgot to add it back.

**Comment by chunnienc** [comment]
@mattsoulanille since this PR introduces some breaking changes, I will not merge it until we release 4.3.1 so that I can have more time to fix them before 4.4.0 . 

Just a note, not pushing us to release 4.3.1.

**Pull request merged**